lib/dao.js

```diff
@@ -2334,6 +2335,24 @@ DataAccessObject.prototype.updateAttributes = function updateAttributes(data, op
     if (err) return cb(err);
     data = ctx.data;
 
+    if (strict) {
+      var props = Model.definition.properties;
+      var keys = Object.keys(data);
+      var result = {};
+      for (var i = 0; i < keys.length; i++) {
+        key = keys[i];
+        if (props[key]) {
+          result[key] = data[key];
+        } else if (strict === 'throw') {
+          cb(new Error('Unknown property: ' + key));
+          return;
+        } else if (strict === 'validate') {
+          inst.__unknownProperties.push(key);
+        } 
+      }
+      data = removeUndefined(result);
+    }
+
```

Comment:
The good news is that models attached to mongodb connector by default have `strict` set to false. For now, maybe we can test `allowExtendedOperators === true` to decide if we should relax the check.